src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -249,6 +250,10 @@ public function addResource(ResourceInterface $resource)
             return $this;
         }
 
+        if ($resource instanceof GlobResource && $this->inVendors($resource->getPrefix())) {
```

Comment:
instead of having a special handling of this resource class here (while others are not filtered), should be have `ContainerBuilder::glob()` like  other methods?